### PR TITLE
feat: Issue queries from querier to ingester in parallel

### PR DESCRIPTION
**Pull request by alamb**
Scratches a personal itch I had (and I think will improve the querier performance on prod): issue the queries to the different ingesters in parallel

End to end test in https://github.com/influxdata/influxdb_iox/pull/4309 covers this code

### CODE REVIEW

File: querier/src/ingester/mod.rs [hunk 1]
```diff
@@ -148,6 +149,120 @@ impl IngesterConnectionImpl {
     }
 }
 
+/// Fetches the partitions for a single ingester
+#[allow(clippy::too_many_arguments)]
```
Comment (alamb) [inline querier/src/ingester/mod.rs:153]: this looks like a large change, but all it does it pull the code from the loop into a function

it is easier to see the difference with a whitespace blind diff in https://github.com/influxdata/influxdb_iox/pull/4359/files?w=1

File: querier/src/ingester/mod.rs [hunk 2]
```diff
@@ -119,7 +120,7 @@ pub trait IngesterConnection: std::fmt::Debug + Send + Sync + 'static {
 #[allow(missing_copy_implementations)]
 #[derive(Debug)]
 pub(crate) struct IngesterConnectionImpl {
-    ingester_addresses: Vec<String>,
+    ingester_addresses: Vec<Arc<str>>,
```
Comment (alamb) [inline querier/src/ingester/mod.rs:123]: This is just to save a string copy -- totally unnecessary I suspect

File: querier/src/ingester/mod.rs [hunk 3]
```diff
@@ -160,112 +279,28 @@ impl IngesterConnection for IngesterConnectionImpl {
...
-                    state.parquet_max_sequence_number.map(SequenceNumber::new),
-                    state.tombstone_max_sequence_number.map(SequenceNumber::new),
-                    batches,
-                )?;
-                ingester_partitions.push(Arc::new(ingester_partition));
-            }
-        }
+                )
+            })
+            .collect::<FuturesUnordered<_>>()
```
Comment (alamb) [inline querier/src/ingester/mod.rs:297]: @domodwyer  taught me this

**Review by crepererum — APPROVED** [review]

File: querier/src/ingester/mod.rs (hunk 3 above)
Comment (domodwyer) [reply querier/src/ingester/mod.rs:297]: I don't remember, but I am glad you like it! 😂

File: querier/src/ingester/mod.rs [hunk 4]
```diff
@@ -148,6 +154,119 @@ impl IngesterConnectionImpl {
...
+#[allow(clippy::too_many_arguments)]
+async fn get_partitions_for_ingester(
+    flight_client: Arc<dyn FlightClient>,
+    catalog_cache: Arc<CatalogCache>,
+    ingester_address: Arc<str>,
+    namespace_name: Arc<str>,
+    table_name: Arc<str>,
+    columns: Vec<String>,
+    predicate: &Predicate,
+    expected_schema: Arc<Schema>,
```
[the remark this answers is not in the document]
Comment (alamb) [reply querier/src/ingester/mod.rs:167]: 👍

File: querier/src/ingester/mod.rs (hunk 3 above)
Comment (alamb) [reply querier/src/ingester/mod.rs:297]: Well, "reading code you wrote taught me this" is perhaps more accurate

File: querier/src/ingester/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (alamb) [reply querier/src/ingester/mod.rs]: in 0015beec9

**Pull request merged**